src/dymoprint/lib/render_engines/print_preview.py

```diff
@@ -0,0 +1,205 @@
+from __future__ import annotations
+
+from PIL import Image, ImageColor, ImageDraw, ImageOps
+
+from dymoprint.lib.render_engines.margins import MarginsMode, MarginsRenderEngine
+from dymoprint.lib.render_engines.render_context import RenderContext
+from dymoprint.lib.render_engines.render_engine import RenderEngine
+from dymoprint.lib.utils import px_to_mm
+
+
+class PrintPreviewRenderEngine(RenderEngine):
+    X_MARGIN_PX = 80
+    Y_MARGIN_PX = 30
+    DX = X_MARGIN_PX * 0.3
+    DY = Y_MARGIN_PX * 0.3
+
+    def __init__(
+        self,
+        render_engine: RenderEngine,
+        justify: str = "center",
+        visible_horizontal_margin_px: float = 0,
+        labeler_margin_px: tuple[float, float] = (0, 0),
+        max_width_px: float | None = None,
+        min_width_px: float = 0,
+    ):
+        super().__init__()
+        self.render_engine = MarginsRenderEngine(
+            render_engine=render_engine,
+            mode=MarginsMode.PREVIEW,
+            justify=justify,
+            visible_horizontal_margin_px=visible_horizontal_margin_px,
+            labeler_margin_px=labeler_margin_px,
+            max_width_px=max_width_px,
+            min_width_px=min_width_px,
+        )
+
+    def _get_label_bitmap(self, context: RenderContext):
+        render_bitmap, meta = self.render_engine.render(context)
+        bitmap = ImageOps.invert(render_bitmap.convert("L")).convert("RGBA")
+        pixel_map = {
+            "black": context.foreground_color,
+            "white": context.background_color,
+        }
+        pixel_map = {
+            ImageColor.getcolor(k, "RGBA"): ImageColor.getcolor(v, "RGBA")
+            for k, v in pixel_map.items()
+        }
+        pixdata = bitmap.load()
+        width, height = bitmap.size
+        for x in range(0, width):
+            for y in range(0, height):
+                pixdata[x, y] = pixel_map[pixdata[x, y]]
+        return bitmap, meta
+
+    def _show_margins(self, label_bitmap, preview_bitmap, meta, context):
```

Comment:
The result looks a bit off:
![image](https://github.com/computerlyrik/dymoprint/assets/15216687/83005fcc-ac2d-4b06-ac60-49ad99c421ec)

The text is white and difficult to read.
The vertical margin appears to be displayed horizontally under the vertical margin.

Reply:
I fixed the coloring. I've hard-coded the colors for now. The reason I've use the original colors is because I worked in dark mode, and it looks different there. I tried to choose colors which would look good on both modes, but this is still a WIP.

> The vertical margin appears to be displayed horizontally under the vertical margin
I am not sure I understand this sentence. Can you elaborate?

Reply:
Ah, now I see. It is really difficult to find something that works in both.

I don't know much about PyQt5, but I wonder if there's some way to set the colors conditionally based on light vs dark.

Reply:
I added a commit which addresses this issue